vgl/pyvgl.cxx

```diff
@@ -708,6 +710,47 @@ void wrap_oriented_box_2d(py::module &m, std::string const& class_name)
     ;
 }
 
+
+template<typename T>
+void wrap_vgl_line_3d_2_points(py::module &m, std::string const& class_name) {
+  bool (*line_3d_2_points_ideal)(const vgl_line_3d_2_points<T>&, T) = &is_ideal;
+  bool (*line_3d_2_points_collinear)(const vgl_line_3d_2_points<T>&, const vgl_point_3d<T>&) = &collinear;
+  bool (*line_3d_2_points_coplanar)(const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&) = &coplanar;
+  bool (*line_3d_2_points_concurrent)(const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&) = &concurrent;
+  bool (*line_3d_2_points_coplanar_2_points)(const vgl_line_3d_2_points<T>&, const vgl_point_3d<T>&, const vgl_point_3d<T>&) = &coplanar;
+  bool (*line_3d_2_points_coplanar_3_lines)(const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&) = &coplanar;
+  bool (*line_3d_2_points_concurrent_3_lines)(const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&,const vgl_line_3d_2_points<T>&) = &concurrent;
+
+  py::class_<vgl_line_3d_2_points<T> > (m, class_name.c_str())
+    .def(py::init<>())
+    .def(py::init<vgl_line_3d_2_points<T>>())
+    .def(py::init<vgl_point_3d<T>, vgl_point_3d<T>>())
+    .def("__len__", [](vgl_line_3d_2_points<T>){return (size_t)2;}) // b/c 2 points? What should this represent
+    // .def("__getitem__", getitem_3d<vgl_line_3d_2_points<T> >) // Is some kind of getitem useful for this class? [0] gets point1 and [1] gets point2 maybe
+    .def("__repr__", streamToString<vgl_line_3d_2_points<T> >)
+    .def(py::pickle(&vslPickleGetState<vgl_line_3d_2_points<T> >,
+                    &vslPickleSetState<vgl_line_3d_2_points<T> >))
+    .def("point1", (vgl_point_3d<T> (vgl_line_3d_2_points<T>::*)(void) const) &vgl_line_3d_2_points<T>::point1)
+    .def("point2", (vgl_point_3d<T> (vgl_line_3d_2_points<T>::*)(void) const) &vgl_line_3d_2_points<T>::point2)
```

Comment:
is it not sufficient to write the following?
```
    .def("point1", &vgl_line_3d_2_points<T>::point1)
    .def("point2", &vgl_line_3d_2_points<T>::point2)
```

Reply:
That would work too. The way it is written now is just an explicit definition of the function. The explicit definition is necessary in cases where there are overloaded functions, but it isn't necessary here